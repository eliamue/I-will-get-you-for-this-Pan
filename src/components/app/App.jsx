import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Detail from '../characters/Detail';

const About = ({ match, location, history }) => {
  return <p>About</p>;
};

export default function App() {
  return (
    <Switch>
      <Route exact path="/:id">
        <Detail />
      </Route>
      <Route exact path="/about" component={About} />
    </Switch>
  );
}
