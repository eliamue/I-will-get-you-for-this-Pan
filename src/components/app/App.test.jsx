import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router-dom';
import rick from '../../data/rick.json';
import charlist from '../../data/charlist.json';
import App from './App';

const server = setupServer(
  rest.get('https://rickandmortyapi.com/api/character/', (req, res, ctx) => {
    return res(ctx.json(charlist));
  }),
  rest.get('https://rickandmortyapi.com/api/character/:id', (req, res, ctx) => {
    return res(ctx.json(rick));
  })
);

describe('App', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('renders list of characters', async () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    screen.getByAltText('Loading');

    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).not.toBeEmptyDOMElement();

    const li = await screen.findByText('Beth Smith');
    expect(li).toBeInTheDocument();
  });

  it('renders a Rick', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/1']}>
        <App />
      </MemoryRouter>
    );

    screen.getByAltText('Loading');

    const character = await screen.findByText('Rick Sanchez');
    expect(character).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
