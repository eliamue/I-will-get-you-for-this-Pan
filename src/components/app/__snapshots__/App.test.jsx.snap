// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a Rick 1`] = `
<div>
  <ul
    aria-label="charlist"
    class="list"
  >
    <li>
      <a
        href="/1"
      >
        <figure
          class="character"
        >
          <img
            alt="Rick Sanchez"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Rick Sanchez
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/2"
      >
        <figure
          class="character"
        >
          <img
            alt="Morty Smith"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Morty Smith
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/3"
      >
        <figure
          class="character"
        >
          <img
            alt="Summer Smith"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Summer Smith
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/4"
      >
        <figure
          class="character"
        >
          <img
            alt="Beth Smith"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Beth Smith
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5"
      >
        <figure
          class="character"
        >
          <img
            alt="Jerry Smith"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Jerry Smith
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/6"
      >
        <figure
          class="character"
        >
          <img
            alt="Abadango Cluster Princess"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Abadango Cluster Princess
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/7"
      >
        <figure
          class="character"
        >
          <img
            alt="Abradolf Lincler"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Abradolf Lincler
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/8"
      >
        <figure
          class="character"
        >
          <img
            alt="Adjudicator Rick"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Adjudicator Rick
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/9"
      >
        <figure
          class="character"
        >
          <img
            alt="Agency Director"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Agency Director
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/10"
      >
        <figure
          class="character"
        >
          <img
            alt="Alan Rails"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Alan Rails
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/11"
      >
        <figure
          class="character"
        >
          <img
            alt="Albert Einstein"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Albert Einstein
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/12"
      >
        <figure
          class="character"
        >
          <img
            alt="Alexander"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Alexander
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/13"
      >
        <figure
          class="character"
        >
          <img
            alt="Alien Googah"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Alien Googah
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/14"
      >
        <figure
          class="character"
        >
          <img
            alt="Alien Morty"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Alien Morty
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/15"
      >
        <figure
          class="character"
        >
          <img
            alt="Alien Rick"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Alien Rick
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/16"
      >
        <figure
          class="character"
        >
          <img
            alt="Amish Cyborg"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Amish Cyborg
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/17"
      >
        <figure
          class="character"
        >
          <img
            alt="Annie"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Annie
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/18"
      >
        <figure
          class="character"
        >
          <img
            alt="Antenna Morty"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Antenna Morty
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/19"
      >
        <figure
          class="character"
        >
          <img
            alt="Antenna Rick"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Antenna Rick
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/20"
      >
        <figure
          class="character"
        >
          <img
            alt="Ants in my Eyes Johnson"
            class="charImage"
            src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
          />
          <figcaption>
            <h2
              class="charName"
            >
              Ants in my Eyes Johnson
            </h2>
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</div>
`;
